src/shared/components/action-buttons/AddToBasket.tsx

```diff
@@ -9,24 +9,37 @@ import accessionToNamespace from '../../utils/accessionToNamespace';
 import { pluralise } from '../../utils/utils';
 import { Namespace } from '../../types/namespaces';
 
-type AddToBasketButtonProps = { selectedEntries: string[] };
+type AddToBasketButtonProps = { selectedEntries: string | string[] };
 
 const AddToBasketButton: FC<AddToBasketButtonProps> = ({ selectedEntries }) => {
-  const [, setBasket] = useBasket();
+  const [basket, setBasket] = useBasket();
 
-  const n = selectedEntries.length;
+  const isSingleEntry = !Array.isArray(selectedEntries);
+
+  const accessions = Array.isArray(selectedEntries)
+    ? selectedEntries
+    : [selectedEntries];
```

Comment:
```suggestion
  const accessions = isSingleEntry
    ? [selectedEntries]
    : selectedEntries;
```

Reply:
I did think about that first, but TS forgets the implication of this boolean when it's used after being set. So at that point, if I were to use that boolean, it would believe `selectedEntries` is `string | string[]` in both places, while re-doing the test in the same statement allows us to have `string` in the first case and `string[]` in the second one.

Reply:
Coming back to this comment as I rememberd this when reading the upcoming features in the next version of TS.
In the future this case should be handled as we both expected it
https://devblogs.microsoft.com/typescript/announcing-typescript-4-4-beta/#cfa-aliased-conditions